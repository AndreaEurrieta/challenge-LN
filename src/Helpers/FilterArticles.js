export const FilterArticles = (data) => {      
        return data.filter(article => article.subtype === "7")
}
