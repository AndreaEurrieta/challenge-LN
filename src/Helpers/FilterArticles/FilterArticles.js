export const FilterArticles = (data) => {
        if (!data || !data.length) return null    
        return data.filter(article => article.subtype === "7")
}
