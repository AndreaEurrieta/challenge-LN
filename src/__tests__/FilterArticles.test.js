import { FilterArticles } from "../Helpers/FilterArticles/FilterArticles";

const articlesMock = [
  {
    _id: "ZNJ67CCHJNAEBE6IUETWOXMNFM",
    display_date: new Date("2019-12-06"),
    headlines: {
      basic: "Arroz con Leche",
    },
    promo_items: {
      basic: {
        url: "https://arc-anglerfish-arc2-sandbox-sandbox-lanacionar.s3.amazonaws.com/public/24QQFAJS3ZGTFL5KXDDUNP5UJU.jpg",
      },
    },
    subtype: "7",
  },
  {
    _id: "ZNJ67CCHJNAEBE6IUETWOXMNFM",
    display_date: new Date("2019-12-06"),
    headlines: {
      basic: "Arroz con Leche",
    },
    promo_items: {
      basic: {
        url: "https://arc-anglerfish-arc2-sandbox-sandbox-lanacionar.s3.amazonaws.com/public/24QQFAJS3ZGTFL5KXDDUNP5UJU.jpg",
      },
    },
    subtype: "2",
  },
];
describe("Helpers-FilterArticles", () => {
  test("Debería devolver solo artículos de subtipo 7", () => {
    const filter = FilterArticles(articlesMock);
    expect(filter).toHaveLength(1);
    expect(filter).toMatchSnapshot();
  });
  test("Debería devolver null si no recibe ningún parametro", () => {
    expect(FilterArticles()).toBeNull();
  });
});
