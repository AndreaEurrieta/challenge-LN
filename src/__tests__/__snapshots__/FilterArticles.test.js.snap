// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helpers-FilterArticles Debería devolver solo artículos de subtipo 7 1`] = `
Array [
  Object {
    "_id": "ZNJ67CCHJNAEBE6IUETWOXMNFM",
    "display_date": 2019-12-06T00:00:00.000Z,
    "headlines": Object {
      "basic": "Arroz con Leche",
    },
    "promo_items": Object {
      "basic": Object {
        "url": "https://arc-anglerfish-arc2-sandbox-sandbox-lanacionar.s3.amazonaws.com/public/24QQFAJS3ZGTFL5KXDDUNP5UJU.jpg",
      },
    },
    "subtype": "7",
  },
]
`;
